kernel/core/datamanagement/memory.cpp

```diff
@@ -3820,6 +3820,226 @@ std::vector<AnovaResult> Memory::getAnova(const VectorIndex& colCategories, size
 }
 
 
+/////////////////////////////////////////////////
+/// \brief check if value type vector is already in a vector of vectors
+///
+/// \param newVector const std::vector<mu::value_type>&
+/// \param centroids const std::vector<std::vector<mu::value_type>>&
+/// \return bool
+///
+/////////////////////////////////////////////////
+static bool isUnique(const std::vector<mu::value_type>& newVector, const std::vector<std::vector<mu::value_type>>& centroids)
+{
+    return std::none_of(centroids.begin(), centroids.end(),
+                        [&newVector](const std::vector<mu::value_type>& centroid)
+                        { return newVector == centroid; });
+}
+
+
+/////////////////////////////////////////////////
+/// \brief Calculate L2 Norm of 2 given value type vectors
+///
+/// \param vec1 const std::vector<mu::value_type>&
+/// \param vec2 const std::vector<mu::value_type>&
+/// \return double
+///
+/////////////////////////////////////////////////
+static double calculateL2Norm(const std::vector<mu::value_type>& vec1, const std::vector<mu::value_type>& vec2)
+{
+
+    mu::value_type sum = 0.0;
+    for (size_t i = 0; i < vec1.size(); ++i)
+    {
+        sum += (vec1[i] - vec2[i]) * (vec1[i] - vec2[i]);
+    }
+
+    // todo do we have sqrt f�r value_type
+    return std::sqrt(sum.real());

```

Comment:
Should work fine for now, because that is only a typedef for `std::complex`. That will change in the near future but... yeah 😉